### PR TITLE
[FLINK-35659][testutils] Fix TestFileSystem Connector error setting execution mode

### PR DESCRIPTION
**Pull request by lsyldliu**
## What is the purpose of the change

*Bugfix for TestFileSystem Connector and optimize code for embedded scheduler*

## Brief change log

  - *Fix TestFileSystem Connector error setting execution mode*
  - *Remove unnecessary parameter customScheduleTime for embedded scheduler*

## Verifying this change

This change is already covered by existing tests.

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): (no)
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: (no)
  - The serializers: (no)
  - The runtime per-record code paths (performance sensitive): (no)
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Kubernetes/Yarn, ZooKeeper: (no)
  - The S3 file system connector: (no)

## Documentation

  - Does this pull request introduce a new feature? (no)
  - If yes, how is the feature documented? (not documented)

### CODE REVIEW

File: flink-test-utils-parent/flink-table-filesystem-test-utils/src/test/java/org/apache/flink/table/file/testutils/TestFileSystemTableFactoryTest.java [hunk 1]
```diff
@@ -64,4 +66,22 @@ void testCreateSourceSink() {
...
+        options.put("source.monitor-interval", "5S");
+
+        DynamicTableSource source = createTableSource(SCHEMA, options);
+        assertThat(source).isInstanceOf(TestFileSystemTableSource.class);
+
+        // assert source is unbounded when specify source.monitor-interval
+        ScanTableSource.ScanRuntimeProvider scanRuntimeProvider =
+                ((TestFileSystemTableSource) source)
+                        .getScanRuntimeProvider(ScanRuntimeProviderContext.INSTANCE);
+        assertThat(scanRuntimeProvider.isBounded()).isFalse();
```
Comment (hackergin) [inline flink-test-utils-parent/flink-table-filesystem-test-utils/src/test/java/org/apache/flink/table/file/testutils/TestFileSystemTableFactoryTest.java:85]: Can we add a new case where, when setting the monitor-interval and the execution mode is batch, it should be a bounded source?

**Review by hackergin — COMMENTED** [review]
@lsyldliu Thanks for your contribution, only one minor comment.

**Comment by lsyldliu** [comment]
@flinkbot run azure

**Review by hackergin — APPROVED** [review]
@lsyldliu Thanks for the update.  LGTM.

**Pull request merged**